Source/Extensions/Blazorise.DataGrid/DataGridState.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Blazorise.DataGrid;
+
+/// <summary>
+/// A DataGrid state container.
+/// </summary>
+/// <typeparam name="TItem"></typeparam>
+public class DataGridState<TItem>
+{
+    /// <summary>
+    /// Gets or sets the current page number.
+    /// </summary>
+    public int CurrentPage { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of items for each page.
+    /// </summary>
+    public int PageSize { get; set; }
+
+    /// <summary>
+    /// Gets the current datagrid editing state.
+    /// </summary>
+    public DataGridEditState EditState { get; set; }
+
+    /// <summary>
+    /// Gets or sets the currently editing item.
+    /// </summary>
+    public TItem EditItem { get; set; }
+
+    /// <summary>
+    /// Gets or sets currently selected row.
+    /// </summary>
+    public TItem SelectedRow { get; set; }
+
+    /// <summary>
+    /// Gets or sets currently selected rows.
+    /// </summary>
+    public List<TItem> SelectedRows { get; set; }
+
+    /// <summary>
+    /// Gets or sets the sort state for the DataGrid columns.
+    /// </summary>
+    public List<DataGridColumnSortState<TItem>> ColumnSortStates { get; set; }
+
+    /// <summary>
+    /// Gets or sets the filter state for the DataGrid columns.
+    /// </summary>
+    public List<DataGridColumnFilterState<TItem>> ColumnFilterStates { get; set; }
+
+    /// <summary>
+    /// Sets the DataGrid to the EditState.New.
+    /// </summary>
+    public void SetNew()
```

Comment:
Rename to `SetNewState()`?

Reply:
It's not a "new state" per say. it's the `EditState` that's set to new. `SetEditModeNew`? `SetEditStateNew`?
Or do you think the semantics wouldn't matter and better keep it short as you say : "`SetNewState`"?

Reply:
It all depends on the context. Here in `SetNew` we set the edit state to `New`. But then, in `SetEdit` method we set the state AND the `editItem`.

We could name them both `SetNewState` and `SetEditItem`, but it would be strange as they both have similar effects, just slight differences. Naming them `SetNewState`, and `SetEditState` would indicate what exactly they do.

Reply:
More or less agree with you, but no point in arguing further, I think it's fine, I'll just make the suggested changes.
Thanks.